flink-formats/flink-orc/src/test/java/org/apache/flink/orc/writer/OrcBulkWriterTest.java

```diff
@@ -75,8 +78,105 @@ public void testOrcBulkWriter() throws Exception {
 
             testHarness.snapshot(1, ++time);
             testHarness.notifyOfCompletedCheckpoint(1);
+        }
+    }
 
-            OrcBulkWriterTestUtil.validate(outDir, input);
+    @Test
+    public void testOrcBulkWriter() throws Exception {
+        final File outDir = TEMPORARY_FOLDER.newFolder();
+        final Properties writerProps = new Properties();
+        writerProps.setProperty("orc.compress", CompressionKind.LZ4.name());
+
+        OrcBulkWriterFactory<Record> writer =
+                new OrcBulkWriterFactory<>(
+                        new RecordVectorizer(schema), writerProps, new Configuration());
+
+        writeRecordsIntoOrcFile(outDir, writer);
+
+        // validate records and compression kind
+        OrcBulkWriterTestUtil.validate(outDir, input, CompressionKind.LZ4);
+    }
+
+    @Test
+    public void testZstdCompression() throws Exception {
+        final File outDir = TEMPORARY_FOLDER.newFolder();
+        final Properties writerProps = new Properties();
+        writerProps.setProperty("orc.compress", CompressionKind.ZSTD.name());
+
+        OrcBulkWriterFactory<Record> writer =
+                new OrcBulkWriterFactory<>(
+                        new RecordVectorizer(schema), writerProps, new Configuration());
+
+        writeRecordsIntoOrcFile(outDir, writer);
+
+        // validate records and compression kind
+        OrcBulkWriterTestUtil.validate(outDir, input, CompressionKind.ZSTD);
```

Comment:
this can be factored out to a private method, given the compression type changes only

Reply:
done